ostd/src/arch/x86/cpu/mod.rs

```diff
@@ -16,13 +16,13 @@ use bitvec::{
     slice::IterOnes,
 };
 use log::debug;
-#[cfg(feature = "intel_tdx")]
+#[cfg(all(target_arch = "x86_64", feature = "intel_tdx"))]
```

Comment:
I think code under `ostd/src/arch/x86` can be put under only `#[cfg(feature = "intel_tdx")]`, since the whole `ostd/src/arch/x86` module is under `#[cfg(target_arch = "x86_64")]`.